lib/Client.js

```diff
@@ -0,0 +1,65 @@
+const { OAuth2RestClient } = require('@elastic.io/component-commons-library');
+
+class Client {
+  constructor(emitter, cfg) {
+    this.logger = emitter.logger;
+    this.cfg = cfg;
+    this.cfg.resourceServerUrl = 'https://graph.microsoft.com/v1.0';
+    this.cfg.authorizationServerTokenEndpointUrl = 'https://login.microsoftonline.com/common/oauth2/v2.0/token';
+    this.cfg.oauth2_field_client_id = process.env.OAUTH_CLIENT_ID;
+    this.cfg.oauth2_field_client_secret = process.env.OAUTH_CLIENT_SECRET;
+    this.restClient = new OAuth2RestClient({ logger: this.logger, emit: emitter.emit }, this.cfg);
+  }
+
+  async getUserInfo() {
+    return this.restClient.makeRequest({
+      url: '/me',
+      method: 'GET',
+    });
+  }
+
+  async getMyMessages() {
+    const response = await this.restClient.makeRequest({
+      url: '/me/messages?$filter=isRead eq false',
+      method: 'GET',
+    });
+
+    return response.value;
+  }
+
+  async getMyContacts(lastModifiedDateTime) {
```

Comment:
The request URI returns only latest contacts. Could we consider to rename it into `getMyLatestContacts`?

Reply:
done